### ISSUE TITLE
[NOT_A_BUG] How fast should memtestCL run?

### ISSUE THREAD

**Issue by birdie-github** [ihaque/memtestCL#7]
I've got a feeling this test is incompatible with NVIDIA Pascal GPUs.

I've got a GTX 1060 GPU, 6096MB of GDDR5 RAM and I get this:

```

./memtestCL 6006
     -------------------------------------------------------------
     |                       MemtestCL v1.00                     |
     |                                                           |
     | Usage: memtestCL [flags] [MB GPU RAM to test] [# iters]   |
     |                                                           |
     | Defaults: GPU 0, 128MB RAM, 50 test iterations            |
     | Amount of tested RAM will be rounded up to nearest 2MB    |
     -------------------------------------------------------------

      Available flags:
        --platform N ,-p N   : run test on the Nth (from 0) OpenCL platform
        --gpu N ,-g N        : run test on the Nth (from 0) OpenCL device
                               on selected platform
        --license ,-l        : show license terms for this build

Error: Bad argument for [MB GPU RAM to test] [# iters]Available OpenCL platforms:
     0: NVIDIA CUDA (SELECTED)
Available OpenCL devices on selected platform:
     0: GeForce GTX 1060 6GB (SELECTED)

Running 50 iterations of tests over 128 MB of memory on device 0: GeForce GTX 1060 6GB

Running memory bandwidth test over 20 iterations of 64 MB transfers...
    Estimated bandwidth 134736.84 MB/s

Test iteration 1 on 128 MiB of memory on device 0 (GeForce GTX 1060 6GB): 0 errors so far
    Moving Inversions (ones and zeros): 0 errors (3 ms)
    Moving Inversions (random): 0 errors (2 ms)
    Memtest86 Walking 8-bit: 0 errors (20 ms)
    True Walking zeros (8-bit): 0 errors (10 ms)
    True Walking ones (8-bit): 0 errors (10 ms)
    Memtest86 Walking zeros (32-bit): 0 errors (41 ms)
    Memtest86 Walking ones (32-bit): 0 errors (40 ms)
    Random blocks: 0 errors (2 ms)
    Memtest86 Modulo-20: 0 errors (51 ms)
    Logic (one iteration): 0 errors (2 ms)
    Logic (4 iterations): 0 errors (9 ms)
    Logic (local memory, one iteration): 0 errors (4 ms)
    Logic (local memory, 4 iterations): 0 errors (11 ms)

... skipped ...

Test summary:
-----------------------------------------
50 iterations over 128 MiB of memory on device GeForce GTX 1060 6GB
      Moving inversions (ones and zeros): 0 failed iterations
                                         (0 total incorrect bits)
                 Memtest86 walking 8-bit: 0 failed iterations
                                         (0 total incorrect bits)
              True walking zeros (8-bit): 0 failed iterations
                                         (0 total incorrect bits)
               True walking ones (8-bit): 0 failed iterations
                                         (0 total incorrect bits)
              Moving inversions (random): 0 failed iterations
                                         (0 total incorrect bits)
             True walking zeros (32-bit): 0 failed iterations
                                         (0 total incorrect bits)
              True walking ones (32-bit): 0 failed iterations
                                         (0 total incorrect bits)
                           Random blocks: 0 failed iterations
                                         (0 total incorrect bits)
                     Memtest86 Modulo-20: 0 failed iterations
                                         (0 total incorrect bits)
                           Integer logic: 0 failed iterations
                                         (0 total incorrect bits)
                 Integer logic (4 loops): 0 failed iterations
                                         (0 total incorrect bits)
            Integer logic (local memory): 0 failed iterations
                                         (0 total incorrect bits)
   Integer logic (4 loops, local memory): 0 failed iterations
                                         (0 total incorrect bits)
Final error count: 0 errors

Press <enter> to quit.
```

All taking roughly **10 seconds**. If we take this GPU bandwidth (134736.84 MB/s) into consideration that means the test could have written and read roughly 134736.84 \* 10 / 2 / 1024 = 657GB +  657GB of data. Now I'm a little bit concerned because a single run of memtest86+ on a PC with dual channel 16GB of RAM DDR3 1600MHz takes around 30 minutes. Of course, my CPU has only 20GB/s bandwidth but it's just 7 times less, which means openCL test should have run for at least four minutes instead of ten seconds.

**Comment by birdie-github** [comment]
It surely looks like the program ignores input data:

> Running 50 iterations of tests over **128 MB** of memory on device 0: GeForce GTX 1060 6GB

**Comment by EoD** [comment]
Apparently it only supports if you specify **both** parameters: ``[MB GPU RAM to test] [# iters]``

See https://github.com/ihaque/memtestCL/blob/master/memtestCL_cli.cpp#L218-L220

**Comment by phly95** [comment]
So testing a GTX 1060 with 6GBs of Vram, what should we enter when running the program?

**Comment by EoD** [comment]
@phly95 What about ``./memtestCL 6000 32`` maybe? Or whatever amount of iterations you want.

**Comment by birdie-github** [comment]
It takes less than a second to complete a single run on my GPU. Something just doesn't feel right - it shouldn't be that fast.

`./memtestCL 6006 1`

**Comment by ihaque** [comment]
Possibly 4GB issue fixed in #13.
